File: mgyoutube-multitech/api-java/src/main/java/com/djpedesen/mgyoutube/api_java/repos/SearchesDataRepo.java
package com.djpedesen.mgyoutube.api_java.repos;

import java.util.List;

public interface SearchesDataRepo {

	void repositoryStartup() throws Exception;

	List<String> getSearchesForParentUser(String parentUserId) throws Exception;

	void addSearchToParentUser(String parentUserId, String searchPhrase) throws Exception;

	void removeSearchFromParentUser(String parentUserId, String searchPhrase) throws Exception;

}

File: mgyoutube-multitech/api-python/repos/SearchesDataRepo.py
from abc import ABC, abstractmethod


class SearchesDataRepo(ABC):
    @abstractmethod
    def repositoryStartup(self):
        pass

    @abstractmethod
    def getSearchesForParentUser(self, parentUserId: str) -> list:
        pass

    @abstractmethod
    def addSearchToParentUser(self, parentUserId: str, searchPhrase: str):
        pass

    @abstractmethod
    def removeSearchFromParentUser(self, parentUserId: str, searchPhrase: str):
        pass
